elyra/pipeline/processor.py

```diff
@@ -148,30 +157,60 @@ def to_json(self):
 
 class PipelineProcessor(LoggingConfigurable):  # ABC
 
-    _type = None
+    _type: str = None
+    _component_registry_location: str = None
 
-    root_dir = Unicode(allow_none=True)
+    _component_parser: ComponentParser = None
+    _component_registry: ComponentRegistry = None
 
-    component_registry: ComponentRegistry = ComponentRegistry()
+    root_dir = Unicode(allow_none=True)
 
     enable_pipeline_info = Bool(config=True,
                                 default_value=(os.getenv('ELYRA_ENABLE_PIPELINE_INFO', 'true').lower() == 'true'),
                                 help="""Produces formatted logging of informational messages with durations
                                 (default=True). (ELYRA_ENABLE_PIPELINE_INFO env var)""")
 
     def __init__(self, root_dir, **kwargs):
-        super(PipelineProcessor, self).__init__(**kwargs)
+        super().__init__(**kwargs)
         self.root_dir = root_dir
 
     @property
     @abstractmethod
-    def type(self):
+    def type(self) -> str:
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def component_registry(self) -> str:
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def component_parser(self) -> ComponentParser:
         raise NotImplementedError()
 
     def get_components(self):
-        components = self.component_registry.get_all_components(processor_type=self.type)
+        """
+        Retrieve components common to all runtimes
+        """
+        components: List[Component] = ComponentRegistry.get_generic_components()
+
+        # Retrieve runtime-specific components
+        if self._component_registry:
+            components.extend(self._component_registry.get_all_components())
+
         return components
 
+    def get_component_properties(self, component_id):
+        """
+        Retrieve runtime-specific component_id details if component_id is not one of the generic set
+        """
+
+        if component_id not in ('notebooks', 'python-script', 'r-script'):
+            return self._component_registry.get_component(component_id=component_id)
```

Comment:
Should this call/method be changed to `get_component_propertes()`?  I'm confused how `get_component()` returns a component's propertes.